tableauserverclient/models/interval_item.py

```diff
@@ -0,0 +1,175 @@
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from .. import NAMESPACE
+
+
+def _validate_time(t):
+    units_of_time = {"hour", "minute", "second"}
+
+    if not any(hasattr(t, unit) for unit in units_of_time):
+        error = "Invalid time object defined."
+        raise ValueError(error)
+
+
+class IntervalItem(object):
+    class Frequency:
+        Hourly = "Hourly"
+        Daily = "Daily"
+        Weekly = "Weekly"
+        Monthly = "Monthly"
+
+    class Occurrence:
+        Hours = "hours"
+        Minutes = "minutes"
+        WeekDay = "weekDay"
+        MonthDay = "monthDay"
+
+    class Day:
+        Sunday = "Sunday"
+        Monday = "Monday"
+        Tuesday = "Tuesday"
+        Wednesday = "Wednesday"
+        Thursday = "Thursday"
+        Friday = "Friday"
+        Saturday = "Saturday"
+        LastDay = "LastDay"
+
+
+class HourlyInterval(object):
+    def __init__(self, start_time, end_time, interval_occurrence, interval_value):
+
+        self.start_time = start_time
+        self.end_time = end_time
+        self.frequency = IntervalItem.Frequency.Hourly
+        self.interval = interval_occurrence.lower(), str(interval_value)
+
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, value):
+        _validate_time(value)
+
+        self._start_time = value
+
+    @property
+    def end_time(self):
+        return self._end_time
+
+    @end_time.setter
+    def end_time(self, value):
+        _validate_time(value)
+        self._end_time = value
+
+    @property
+    def interval(self):
+        return self._interval
+
+    @interval.setter
+    def interval(self, intervals):
+        interval_occurrence, interval_value = intervals
+
+        Hours = IntervalItem.Occurrence.Hours
+        Minutes = IntervalItem.Occurrence.Minutes
+
+        VALID_HOUR_INTERVALS = {1, 2, 4, 6, 8, 12}
+        VALID_MIN_INTERVALS = {15, 30}
+
+        if interval_occurrence not in {Hours, Minutes}:
+            error = "Invalid interval type defined: {}.".format(interval_occurrence)
+            raise ValueError(error)
+        elif interval_occurrence == Hours and int(interval_value) not in VALID_HOUR_INTERVALS:
```

Comment:
why are we casting to int? The user should pass in an int. I would rather have clear types instead of let's see what happens. If they didn't give us an int ... error